pkg/controller/engine.go

```diff
@@ -271,6 +271,10 @@ func (c *namedController) Start(ctx context.Context) error {
 	}()
 	go func() {
 		<-c.e.mgr.Elected()
+		if synced := c.ca.WaitForCacheSync(ctx); !synced {
+			c.e.done(c.name, errors.New(errCrashCache))
+			return
+		}
```

Comment:
I think this is already happening inside of controller-runtime for the non-realtime case. I find their code hard to follow due to all the returning of interfaces, but AFAICT:

1. Our `e.Create` calls `ctrl.Watch`, with `source.Kind(ca, ...)` as an argument. This adds the source to the controller's internal `startWatches` slice of watches to start when `ctrl.Start` is called.
2. When `ctrl.Start` is called it iterates over `startWatches`, calling each source's `Start` method.
3. The source should be an `internal.Kind`, and its `Start` method calls its cache's `WaitForCacheSync` method.

That's not guaranteed to be the case when using `TriggeredBy` though.

Assuming it doesn't hurt to call `WaitForCacheSync` multiple times I think it's a good idea to add this regardless.